code/controllers/subsystem/garbage.dm

```diff
@@ -207,6 +210,14 @@ var/datum/subsystem/garbage_collector/SSgarbage
 		return TRUE
```

Comment:
They should return False, that's the expected behaviour due to legacy implementation, that's why I added a new proc, to avoid breaking the api in anyway other than disallowing repeated calls to destroy.

Basically the object is only qdeleted when it's actually qdeleted.

An argument could be made to change how it behaves but I'm not particularly bothered.